kolibri/core/tasks/test/test_job.py

```diff
@@ -364,7 +364,24 @@ def test_enqueue(self, job_storage_mock, _ready_job_mock):
             priority=self.registered_task.priority,
             retry_interval=None,
         )
+    @mock.patch("kolibri.core.tasks.registry.RegisteredTask._ready_job")
+    @mock.patch("kolibri.core.tasks.registry.job_storage")
+    def test_enqueue_lifo_job(self, job_storage_mock, _ready_job_mock):
+        args = ("10",)
+        kwargs = dict(base=10)
+
+        _ready_job_mock.return_value = "lifo_job"
+
+        self.registered_task.enqueue_lifo_job(args=args, kwargs=kwargs)
 
+        _ready_job_mock.assert_called_once_with(args=args, kwargs=kwargs)
+        job_storage_mock.enqueue_lifo.assert_called_once_with(
+            "lifo_job",
+            queue=self.registered_task.queue,
+            priority=self.registered_task.priority,
+            retry_interval=None,
+        )
+        
```

Comment:
We should also add a test to the test_storage test suite that confirms the LIFO behaviour, first enqueueing multiple jobs as LIFO, and then ensuring that the last queued job is that returned by `get_next_queued_job`.

If you add this test, you should find it will fail, because the `_filter_next_query` method does not currently order by `scheduled_time`:

```
    def _filter_next_query(self, query, priority):
        naive_utc_now = datetime.utcnow()
        return (
            query.filter(ORMJob.state == State.QUEUED)
            .filter(ORMJob.scheduled_time <= naive_utc_now)
            .filter(ORMJob.priority <= priority)
            .order_by(ORMJob.priority, ORMJob.time_created)
        )
```

Updating it to:
```
    def _filter_next_query(self, query, priority):
        naive_utc_now = datetime.utcnow()
        return (
            query.filter(ORMJob.state == State.QUEUED)
            .filter(ORMJob.scheduled_time <= naive_utc_now)
            .filter(ORMJob.priority <= priority)
            .order_by(ORMJob.priority, ORMJob.scheduled_time, ORMJob.time_created)
        )
```

Should fix the as yet unwritten test. I would suggest writing the new test first, confirming that it fails, and then making this modification to be sure it fixes it.